File: data/gen.rb
#!/usr/bin/env ruby

require './translit.rb'
require 'json'

debug = ARGV[0] == "-d"

count = 0

puts "Parsing JSON"

doc = JSON.parse(File.read("stops.json"))
sql = ""

puts "Parsed JSON"

doc["Stops"].each do |node|
	next if node["Type"] != "bus"
	id = node["StopId"]
	name = node["Name"]
	name = name.split(" - ")[0]
	name_en = translit name
	lat = node["Lat"]
	lon = node["Lon"]
	has_board = (node["HasBoard"] == "true") ? 1 : 0
	has_data = (node["Virtual"] == "true") ? 1 : 0
    
    sql_record = "INSERT INTO stops(id,name,name_en,lat,lon,hasData,hasBoard) \
VALUES(#{id},'#{name}','#{name_en}',#{lat},#{lon},#{has_data},#{has_board});\n"
    sql += sql_record

	puts "#{id} #{name} #{name_en}" if debug
	puts sql_record if debug
	count += 1
end

puts "Processed #{count} records"
puts "Writing SQL file"
File.open "db.sql", "w+" do |file|
	file.write sql
end
puts "Success! db.sql generated."

File: data/gen.js
#!/usr/bin/env node

var fs = require('fs'),
  data = fs.readFileSync("stops.json"),
  data_en = fs.readFileSync("stops-en.json"),
  json_data = JSON.parse(data),
  json_data_en = JSON.parse(data_en),
  stops_map = {},
  out_data = [];

for (var i in json_data.Stops) {
  stops_map[json_data.Stops[i].StopId] = json_data.Stops[i];
}
for (var i in json_data_en.Stops) {
  var stop = json_data_en.Stops[i];
  if (stops_map[stop.StopId]) {
    stops_map[stop.StopId].Name_en = stop.Name;
  }
}

for (var i in stops_map) {
  var stop = stops_map[i];

  if (stop.Type != "bus" || !parseInt(stop.StopId)) continue;

  var name = stop.Name.replace(/ ?- \[[0-9]+\]/ig, "");
  console.log(name);

  out_data.push({
    lat: parseFloat(stop.Lat),
    lon: parseFloat(stop.Lon),
    name: name,
    name_en: stop.Name_en,
    id: stop.StopId
  });
}

fs.writeFileSync("db.json", JSON.stringify(out_data));
